### PR TITLE
Feature/interface switch chain

### PR DESCRIPTION
**Pull request by ahwei**
# Description

feat:interface switch chain
scan

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/api/extension/messaging.ts [hunk 1]
```diff
@@ -53,6 +53,12 @@ class Messaging {
             if (isValidWalletRes?.data?.error) {
                 throw new Error(isValidWalletRes.data.message)
             }
+            // before enbale need to check unlock
```
Comment (hryjosn) [inline src/api/extension/messaging.ts]: before enable need to check if unlock

File: src/api/extension/webpage.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import Messaging from '~/api/extension/messaging'
-import { MethodEnum } from '~/types/extension'
+import { MethodEnum, RequestMethodEnum } from '~/types/extension'
 
 export const enable = async () => {
+    console.log('start enable')
```
Comment (hryjosn) [inline src/api/extension/webpage.ts]: Remove console.log

File: src/background/index.ts [hunk 3]
```diff
@@ -140,4 +140,46 @@ controller.add(MethodEnum.RESET_PASSWORD, async (request) => {
     } catch (error) {}
 })
 
+controller.add(MethodEnum.IS_LOCK, async (request, sendResponse) => {
+    const mnemonic = await getStorage<string>('mnemonic')
+    const password = controller?.password
+    const isLocked = !password && !mnemonic
```
Comment (hryjosn) [inline src/background/index.ts:152]: Remove mnemonic condition

File: src/injected-scripts/index.ts [hunk 4]
```diff
@@ -1,10 +1,31 @@
...
 
+import pkg from '../../package.json'
 window.chia = {
     ...(window.chia || {}),
     hoogii: {
         name: 'Hoogii',
         apiVersion: '0.0.4',
+        version: pkg.version,
+        isHoogii: true,
+        request: async (arg: { method: RequestMethodEnum }) => {
```
Comment (hryjosn) [inline src/injected-scripts/index.ts]: if (res?.data.error) {
    throw res?.data
}
return res?.data

File: src/popup/controller.ts [hunk 5]
```diff
@@ -19,6 +18,7 @@ export class InternalControllerStore {
     private tabId?: number
     request?: IMessage
     locked = true
+    connected = false
```
Comment (hryjosn) [inline src/popup/controller.ts:22]: remove connected state

File: src/popup/controller.ts [hunk 6]
```diff
@@ -84,6 +86,20 @@ export class InternalControllerStore {
             tabId: this.tabId,
         } as InternalReturnType)
     }
+
+    connectSite = async () => {
```
Comment (hryjosn) [inline src/popup/controller.ts]: This function should change to request indexDb

File: src/popup/controller.ts [hunk 7]
```diff
@@ -84,6 +86,20 @@ export class InternalControllerStore {
...
+        runInAction(() => {
+            this.connected = true
+        })
+    }
+
+    onFinishRequest = async (response = true) => {
```
Comment (hryjosn) [inline src/popup/controller.ts]: Rename to resposnseFromBackground

File: src/popup/controller.ts [hunk 8]
```diff
@@ -84,6 +86,20 @@ export class InternalControllerStore {
...
+        // response is to background script
```
Comment (hryjosn) [inline src/popup/controller.ts]: request to background script

**Comment by ahwei** [comment]
TODO FIX
- [x] connect using indexDB
- [x] rebuild onFinishRequest
- [x] use functionial to handle create popup
- [x] fix connect auth bug

File: src/popup/index.tsx [hunk 9]
```diff
@@ -27,7 +27,7 @@ const App = observer(() => {
         if (request && request.method === MethodEnum.REFUSE) {
             navigate('/refuse')
         }
-
+        console.log('locked>', locked, 'connected>', connected)
```
Comment (hryjosn) [inline src/popup/index.tsx]: remove console.log

File: src/components/Item.tsx [hunk 10]
```diff
@@ -133,10 +133,16 @@ export const ConnectedSiteItem = ({
...
     iconUrl,
     action,
+    href,
     ...rest
 }: IConnectedSiteItemProps & ComponentProps<typeof Item<'a'>>) => (
-    <Item as="a" key={name} {...rest} className="!cursor-pointer">
-        <span className="gap-2 flex-row-center">
+    <Item key={name} {...rest} className="!cursor-pointer">
+        <span
+            className="gap-2 flex-row-center"
```
Comment (hryjosn) [inline src/components/Item.tsx]: revert this part of code

File: src/api/extension/request.ts [hunk 11]
```diff
@@ -48,12 +47,12 @@ const doubleCheckHandler = async (request: IMessage<RequestArguments>) => {
...
 export const requestHandler = async (request: IMessage<RequestArguments>) => {
-    const check = await doubleCheckHandler(request)
-    if (!check) {
+    const confirmed = await requestConfirmHandler(request)
+    if (!confirmed) {
         return {
             error: true,
             code: 4002,
-            message: 'user rejected request',
+            message: 'user confirmed reject',
```
Comment (hryjosn) [inline src/api/extension/request.ts]: reject the request

File: src/api/extension/request.ts [hunk 12]
```diff
@@ -52,7 +52,7 @@ export const requestHandler = async (request: IMessage<RequestArguments>) => {
         return {
             error: true,
             code: 4002,
-            message: 'user confirmed reject',
+            message: 'user rejected confirm',
```
Comment (hryjosn) [inline src/api/extension/request.ts]: Reject the request

**Review by hryjosn — APPROVED** [review]

**Pull request merged**